http-native/src/main/java/org/ballerinalang/net/http/actions/websocketconnector/Close.java

```diff
@@ -107,23 +103,25 @@ private static void waitForTimeout(NonBlockingCallback callback, int timeoutInSe
                     String errMsg = String.format(
                             "Could not receive a WebSocket close frame from remote endpoint within %d seconds",
                             timeoutInSecs);
-                    setReturnValues(errMsg, callback);
+                    addError(errMsg);
                     WebSocketObservabilityUtil.observeError(connectionInfo,
                                                             WebSocketObservabilityConstants.ERROR_TYPE_CLOSE, errMsg);
                 }
             }
         } catch (InterruptedException err) {
             String errMsg = "Connection interrupted while closing the connection";
-            setReturnValues(errMsg, callback);
+            addError(errMsg);
             WebSocketObservabilityUtil.observeError(connectionInfo,
                                                     WebSocketObservabilityConstants.ERROR_TYPE_CLOSE, errMsg);
             Thread.currentThread().interrupt();
         }
     }
 
-    private static void setReturnValues(String errMsg, NonBlockingCallback callback) {
-        callback.setReturnValues(WebSocketUtil.getWebSocketException(errMsg, null,
-                ErrorCode.WsConnectionClosureError.errorCode(), null));
+    private static void addError(String errMsg) {
+//        TODO: error is not getting thrown fix this
+//
+//        WebSocketException error = WebSocketUtil.getWebSocketException(errMsg, null,
```

Comment:
```suggestion
//        throw WebSocketUtil.getWebSocketException(errMsg, null, ErrorCode.WsConnectionClosureError.errorCode(), null);
```

Reply:
Since this is a Async method, throwing is not the best option, I have don't it the correct way in #50 but there are issues in the tests those have to be fixed before fixing this.

Reply:
Ack.

Reply:
This is tracked in https://github.com/ballerina-platform/module-ballerina-http/issues/59